cmd/srcd/cmd/components.go

```diff
@@ -145,10 +135,71 @@ var componentsInstallCmd = &cobra.Command{
 	},
 }
 
+// componentsStartCmd represents the components start command
+var componentsStartCmd = &cobra.Command{
+	Use:   "start component [component...]",
+	Short: "Start source{d} component",
+	Args:  cobra.MinimumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := daemon.Client()
+		if err != nil {
+			return humanizef(err, "could not get daemon client")
+		}
+
+		cmps, err := components.List(context.Background(), false)
+		if err != nil {
+			return humanizef(err, "could not list images")
+		}
+
+		for _, arg := range args {
+			c, err := getComponent(arg, cmps)
+			if err != nil {
+				return err
+			}
+
+			ctx := context.Background()
+			started := logAfterTimeoutWithServerLogs("this is taking a while, "+
+				"it might take a few more minutes while we install all the required images",
+				5*time.Second)
+			_, err = client.StartComponent(ctx, &api.StartComponentRequest{
+				Name: c.Name,
+			})
+			started()
+			if err != nil {
+				return humanizef(err, "could not start %s", c.Name)
+			}
+		}
+
+		return nil
+	},
+}
+
+func getComponent(arg string, cmps []components.Component) (*components.Component, error) {
+	var c *components.Component
+	for _, cmp := range cmps {
+		if arg == cmp.Name || arg == cmp.Image {
```

Comment:
It would be so nice that we could also let the user ask for the shorter version of the component name...
it seems that `srcd-cli-` is a common and static thing, so we could also add `"srcd-cli-"+arg == cmp.Name`

Reply:
yeah. I think we need this feature. But I'm not sure about this particular implementation. Maybe it's better to expose correct names in the components. I have created an issue: https://github.com/src-d/engine/issues/438

Reply:
I totally agree with you. Opening a separate issue was a good decision, so thanks.